### PR TITLE
refactor: Remove the depends on settings from license check

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR will remove the depends on settings from license check.

## Notews

`common-settings` heavily relies on `common-config` and `common-meta-app`. This dependency creates a cycle for `Settings`. This will block `common-storage` to depend on `common-license`.

This PR resolves this issue by directly passing the license key instead of accepting an `Arc<Settings>`. Additionally, since our license check doesn't utilize the `tenant` (it's only used for logging), I've removed it as well.

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13343)

### CODE REVIEW

File: src/bendpy/src/lib.rs [hunk 1]
```diff
@@ -57,7 +57,7 @@ fn databend(_py: Python, m: &PyModule) -> PyResult<()> {
         GlobalServices::init(conf.clone()).await.unwrap();
 
         // init oss license manager
-        OssLicenseManager::init().unwrap();
+        OssLicenseManager::init(conf.query.tenant_id.clone()).unwrap();
```
Comment (BohuTANG) [inline src/bendpy/src/lib.rs:60]: This line code format looks not pretty.

Comment (Xuanwo) [reply src/bendpy/src/lib.rs:60]: > This line code format looks not pretty.

Are you talking about refactor code here into:

```rust
let tenant_id = conf.query.tenant_id.clone();

OssLicenseManager::init(tenant_id).unwrap();
```

**Comment by Xuanwo** [comment]
CI failed for

```shell
0: query failed: Databend sqllogictests error: http query error: code: 1067, Text: transport error, source: Some(tonic::transport::Error(Transport, hyper::Error(Io, Kind(ConnectionReset))))
(while in query flight)
(while in query flight)
Error: SelfError("sqllogictest failed")
[SQL] select number, and_filters(number > 1, number > 2 , number > 3) from numbers(6) order by number;
at tests/sqllogictests/suites/query/02_function/02_0046_function_logic:309
```

I'm guessing it's not related to this PR?

**Comment by BohuTANG** [comment]
Yes, I have re-run the failed CI.

**Review by ZhiHanZ — APPROVED** [review]

**Pull request merged**